step-controller/step-controller-server/src/main/java/step/core/scheduler/Executor.java

```diff
@@ -127,12 +142,45 @@ public boolean schedule(ExecutiontTaskParameters task) {
 			logger.error("An error occurred while checking if task exists in scheduler: " + task);
 			throw new RuntimeException(e);
 		}
-		Trigger trigger = TriggerBuilder.newTrigger().withSchedule(CronScheduleBuilder.cronSchedule(task.getCronExpression())).build();
+		//Base calendar allows you to chain calendars.
+		BaseCalendar baseCalendar = new BaseCalendar();
+		if (task.getCronExclusions() != null) {
+			try {
+				for(CronExclusion c : task.getCronExclusions()) {
+					baseCalendar = new CronCalendar(baseCalendar, c.getCronExpression());
+				}
+			} catch (ParseException e) { //such exception should already be caught by validate
+				logAndThrow("One of the cron expressions for the task '" + task.getAttribute(AbstractOrganizableObject.NAME) +
+						"' is invalid.", e);
+			}
+		}
+		//In case exclusions were removed we need to add/replace at least the base calendar
+		try {
+			scheduler.addCalendar("exclusionsCalendar_" + task.getId().toHexString(), baseCalendar, true, false);

```

Comment:
Just for my understanding: where do we specify that this calendar is an exclusion calendar? is it implicit in the method modifiedByCalendar?

Reply:
@jeromecomte in the Trigger builder ".modifiedByCalendar"